cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -372,26 +379,77 @@ impl ClientTrait for CatalogClient {
     }
 }
 
+/// Collects a stream of search results into a container, returning the total count as well.
+///
+/// Note: it is assumed that the first element of the stream contains the total count.
+async fn collect_search_results<T, E>(
```

Comment:
suggestion nonblocking: I still find streams kinda confusing, so it feels simpler to me if we just have one function that returns a `Vec<T>` than one that returns a stream and one that collects it

Reply:
I hear you, but I prefer it this way for two reasons:
- Producing the stream and consuming the streams are two independent operations
- Having the stream-consuming logic separate allows you to test it with any stream (e.g. one produced via `proptest`) instead of being forced to use the one that's generated by consuming pages.

Reply:
Async Rust takes a lot of getting used to, which is why I've avoided it inside `flox` for so long, but now that we're stuck with it (because of the catalog client) we might as well use the best practices.